src/client/AM.PA.MacOs/PersonalAnalytics/Trackers/WindowsActivityTracker/ActivitySQLController.swift

```diff
@@ -118,26 +98,28 @@ class ActivitySQLController: SQLController{
     }
     
     func GetLongestFocusOnProgram(date: Date) -> FocusedWorkDict?{
-        let start = getStartHour(date:date)
-        let end = getEndHour(date:date)
+        let startStr = DateFormatConverter.interval1970ToDateStr(interval: getStartHour(date:date))
+        let endStr = DateFormatConverter.interval1970ToDateStr(interval: getEndHour(date:date))
         
         do{
-            var query = "SELECT *, ZENDTIME - ZSTARTTIME AS DIFF FROM ZACTIVEAPPLICATION "
-            query += "WHERE ZSTARTTIME >= " + String(start)
-            query += " AND ZENDTIME <= " + String(end)
-            query += " AND ZNAME <> 'Idle' "
-            query += "ORDER BY DIFF DESC "
-            query += "LIMIT 1;"
+            let query = """
+                        SELECT process, tsStart, tsEnd, (strftime('%s', tsEnd) - strftime('%s', tsStart)) AS diff
+                        FROM windows_activity
+                        WHERE tsStart >= '\(startStr)' AND tsEnd <= '\(endStr)' AND process <> 'Idle'
+                        GROUP BY id, tsStart
+                        ORDER BY diff DESC
+                        LIMIT 1
+                        """
             
             let row = try dbQueue.inDatabase{db in
-                try Row.fetchOne(db, query)
+                try Row.fetchOne(db, sql: query)
             }
             
             if(row != nil){
-                let name: String = (row?["ZNAME"])!
-                let from:TimeInterval = (row?["ZSTARTTIME"])!
-                let to:TimeInterval = (row?["ZENDTIME"])!
-                let difference:TimeInterval = (row?["DIFF"])!
+                let name: String = row!["process"]
+                let from:TimeInterval = DateFormatConverter.dateStrToInterval1970(str: row!["tsStart"])
+                let to:TimeInterval = DateFormatConverter.dateStrToInterval1970(str: row!["tsEnd"])
+                let difference:TimeInterval = row!["diff"]
```

Comment:
We should probably add constants for these field names (tsStart, tsEnd, etc..) somewhere as well as for the table names.

Reply:
Good point! Noted.